File: models/src/main/java/com/mdgd/pokemon/models/filters/FilterData.java
package com.mdgd.pokemon.models.filters;

import java.util.ArrayList;
import java.util.List;

public class FilterData {
    public static final String FILTER_ATTACK = "attack";
    public static final String FILTER_DEFENCE = "defense";
    public static final String FILTER_SPEED = "speed";

    private final List<String> filters;

    public FilterData() {
        this.filters = new ArrayList<>(0);
    }

    public FilterData(List<String> filters) {
        this.filters = filters;
    }

    public List<String> getFilters() {
        return filters;
    }

    public boolean isEmpty() {
        return filters.isEmpty();
    }
}

File: models/src/main/java/com/mdgd/pokemon/models/filters/FilterData.kt
package com.mdgd.pokemon.models.filters

import java.util.*

class FilterData {
    val filters: List<String>

    constructor() {
        filters = LinkedList()
    }

    constructor(filters: List<String>) {
        this.filters = filters
    }

    val isEmpty: Boolean
        get() = filters.isEmpty()

    companion object {
        const val FILTER_ATTACK = "attack"
        const val FILTER_DEFENCE = "defense"
        const val FILTER_SPEED = "speed"
    }
}
